Platform/ElkhartlakeBoardPkg/Script/StitchIfwiConfig.py

```diff
@@ -72,15 +72,26 @@ def get_platform_sku():
     }
     return platform_sku
 
-def get_component_replace_list():
+def get_component_replace_list(work_dir):
+    ipfw = os.path.join(work_dir, 'IPFW')
+    print ("\nIPFW dir : %s" % ipfw)
+    CompFilePreOsChecker='IPFW/PreOsChecker.bin' if os.path.exists(os.path.join(ipfw, 'PreOsChecker.bin')) else ''
+    CompFilePseFw='IPFW/PseFw.bin' if os.path.exists(os.path.join(ipfw, 'PseFw.bin')) else ''
+    CompFilePseTsnIpConfig='IPFW/PseTsnIpConfig.bin' if os.path.exists(os.path.join(ipfw, 'PseTsnIpConfig.bin')) else ''
+    CompFileTsnConfig='IPFW/TsnConfig.bin' if os.path.exists(os.path.join(ipfw, 'TsnConfig.bin')) else ''
+    CompFileTsnMacAddr='IPFW/TsnMacAddr.bin' if os.path.exists(os.path.join(ipfw, 'TsnMacAddr.bin')) else ''
+    CompFileTccCrl='IPFW/crl.bin' if os.path.exists(os.path.join(ipfw, 'crl.bin')) else ''
+
     replace_list = [
-      ('IFWI/BIOS/TS0/ACM0',      'Input/acm.bin',          'dummy',    '', ''),
-      ('IFWI/BIOS/TS1/ACM0',      'Input/acm.bin',          'dummy',    '', ''),
-      ('IFWI/BIOS/NRD/IPFW/POSC', 'IPFW/PreOsChecker.bin',   'dummy',   'KEY_ID_CONTAINER_COMP_RSA3072', 0),   # Pre-OS Checker
-      ('IFWI/BIOS/NRD/IPFW/PSEF', 'IPFW/PseFw.bin',          'lz4',     'KEY_ID_CONTAINER_COMP_RSA3072', 0),   # OSE FW
-      ('IFWI/BIOS/NRD/IPFW/TSIP', 'IPFW/PseTsnIpConfig.bin', 'lz4',     'KEY_ID_CONTAINER_COMP_RSA3072', 0),   # PSE TSN IP
-      ('IFWI/BIOS/NRD/IPFW/TSNC', 'IPFW/TsnConfig.bin',      'lz4',     'KEY_ID_CONTAINER_COMP_RSA3072', 0),   # TSN Config
-      ('IFWI/BIOS/NRD/IPFW/TMAC', 'IPFW/TsnMacAddr.bin',     'lz4',     'KEY_ID_CONTAINER_COMP_RSA3072', 0),   # TSN MAC Address
+       #    Path                   file name              compress    Key                           SVN
+      ('IFWI/BIOS/TS0/ACM0',      'Input/acm.bin',        'dummy',    '',                            ''),
+      ('IFWI/BIOS/TS1/ACM0',      'Input/acm.bin',        'dummy',    '',                            ''),
+      ('IFWI/BIOS/NRD/IPFW/POSC', CompFilePreOsChecker,   'dummy',   'KEY_ID_CONTAINER_COMP_RSA3072', 0),   # Pre-OS Checker
+      ('IFWI/BIOS/NRD/IPFW/PSEF', CompFilePseFw,          'lz4',     'KEY_ID_CONTAINER_COMP_RSA3072', 0),   # OSE FW
+      ('IFWI/BIOS/NRD/IPFW/TSIP', CompFilePseTsnIpConfig, 'lz4',     'KEY_ID_CONTAINER_COMP_RSA3072', 0),   # PSE TSN IP
+      ('IFWI/BIOS/NRD/IPFW/TSNC', CompFileTsnConfig,      'lz4',     'KEY_ID_CONTAINER_COMP_RSA3072', 0),   # TSN Config
+      ('IFWI/BIOS/NRD/IPFW/TMAC', CompFileTsnMacAddr,     'lz4',     'KEY_ID_CONTAINER_COMP_RSA3072', 0),   # TSN MAC Address
+      ('IFWI/BIOS/NRD/IPFW/TCCM', CompFileTccCrl,         'lz4',     'KEY_ID_CONTAINER_COMP_RSA3072', 0),   # TCC CRL binary
     ]

```

Comment:
Check if you use below code to avoid changing StitchIfwi.py since we would like to have a common StitchIfwi.py late:
    replace_list = [
      ('IFWI/BIOS/TS0/ACM0',      'Input/acm.bin',          'dummy',    '', ''),
      ('IFWI/BIOS/TS1/ACM0',      'Input/acm.bin',          'dummy',    '', ''),
    ]
    if os.path.exists('IPFW/PreOsChecker.bin'):
            replace_list.append (
              ('IFWI/BIOS/NRD/IPFW/POSC', 'IPFW/PreOsChecker.bin',   'dummy',   'KEY_ID_CONTAINER_COMP_RSA3072', 0),   # Pre-OS Checker
            )

Reply:
love this, this is really neat. Thanks! Done.